src/components/StatusIcon.vue

```diff
@@ -0,0 +1,25 @@
+<template>
+  <div class="status-icon" :class="`status-icon--${status}`" />
+</template>
+
+<script setup lang="ts">
+  export type Status = 'ready' | 'not_ready'
```

Comment:
I considered having this be more generic like a `variant='positive' | 'negative'` or so but after work queue health -> status, the only semantic mismatch is Worker which uses "online" vs "offline". everything else is using "ready" | "not_ready" semantics so figured this works well for most cases and just Worker can pivot (see changes to `WorkerStatusBadge`)